File: keepr/server/Services/ProfilesService.cs
using Keepr.Models;

namespace keepr.Services;

public class ProfilesService
{
    private readonly ProfilesRepository _profilesRepository;

    public ProfilesService(ProfilesRepository profilesRepository)
    {
        _profilesRepository = profilesRepository;
    }

    public Profile GetProfileById(string profileId)
    {
        Profile profile = _profilesRepository.GetProfileById(profileId);
        if (profile == null)
        {
            throw new Exception("Profile not found");
        }
        return profile;
    }

    internal List<Keep> GetKeepsByProfileId(string profileId)
    {
        List<Keep> keeps = _profilesRepository.GetKeepsByProfileId(profileId);
        return keeps;
    }

    internal List<Vault> GetVaultsByProfileId(string profileId, string userId)
    {
        List<Vault> vaults = _profilesRepository.GetVaultsByProfileId(profileId);

        vaults = vaults.Where(vault => !vault.IsPrivate || vault.CreatorId == userId).ToList(); //Filter out private vaults not owned by the requesting user
        return vaults;
    }
}

File: keepr/client/src/services/ProfilesService.js
import { AppState } from "@/AppState.js"
import { Profile } from "@/models/Account.js"
import { api } from "./AxiosService.js"
import { Keep } from "@/models/Keep.js";
import { Vault } from "@/models/Vault.js";


class ProfilesService {
  async getVaultsByProfileId(profileId) {
    console.log('Fetching vaults for profile ID:', profileId);
    const response = await api.get(`api/profiles/${profileId}/vaults`);
    
    console.log('getVaultsByProfileId res', response);
    AppState.vaults = response.data.map(vaultData => new Vault(vaultData));
  }

  async getKeepsByProfileId(profileId) {
    console.log('Fetching keeps for profile ID:', profileId);

    const response = await api.get(`api/profiles/${profileId}/keeps`);
    
    console.log('getKeepsByProfileId res', response);
    AppState.keeps = response.data.map(keepData => new Keep(keepData));
  }

  async getProfileByProfileId(profileId) {
    console.log('Fetching profile with ID:', profileId);
    const response = await api.get(`api/profiles/${profileId}`);

    console.log('getProfileByProfileId res', response);
    AppState.profile = new Profile(response.data)
  }
}

export const profilesService = new ProfilesService()